File: #0165.compare-version-numbers.cpp
class Solution {
public:
    int compareVersion(string version1, string version2) {

        int v1size = version1.size();
        int v2size = version2.size();
        int i = 0;
        int j = 0;
        int v1 = 0;
        int v2 = 0;
        while (i < v1size || j < v2size) {
            while (i < v1size && version1[i] != '.') {
                v1 = v1 * 10 + version1[i] - '0';
                i++;
            }

            while(j < v2size && version2[j] != '.') {
                v2 = v2 * 10 + version2[j] - '0';
                j++;
            }

            if (v1 > v2) {
                return 1;
            }
            
            if (v1 < v2) {
                return -1;
            }

            v1 = 0;
            v2 = 0;
            i++;
            j++;
        }
        return 0;
    }
};

File: #0165.compare-version-numbers.js
/**
 * @param {string} version1
 * @param {string} version2
 * @return {number}
 */
var compareVersion = function(version1, version2) {
    const [arr1, arr2] = [version1.split('.'), version2.split('.')];
    const length = Math.max(arr1.length, arr2.length);
    for (let i = 0; i < length; i++) {
        const rev1 = parseInt(arr1[i]) || 0;
        const rev2 = parseInt(arr2[i]) || 0;
        if (rev1 < rev2) {
            return -1;
        }

        if (rev1 > rev2) {
            return 1;
        }
    }
    return 0;
};
